controllers/Admin.js

```diff
@@ -242,96 +242,164 @@ router.post('/images/upload', upload.array('file', 20), async ctx => {
 // new products
 
 router.get('/product', function (req, res) {
   const product = {}
   res.render('admin/product', { product })
 })
 
-// view/edit projects
-router.get('/product/:id', function (req, res) {
-  var id = req.params.id
-  Product.findOne({ _id: id }, function (err, product) {
-    res.render('admin/product', {
-      product
-    })
-  })
+/*** KOA /product route ********/
+router.get('/product', async ctx => {
+  const product = {}
+  const query = ctx.query
+
+  _.merge(product, query)
+  ctx.render('admin/product', { product })
 })
 
-// add new/edit
-router.post('/product', function (req, res) {
-  var id = req.body._id
-  var body = req.body
+/****** End of KOA /product route ********/
 
-  var errors = []
-  if (!validator.isCurrency(body.price))
-    errors.push('Price is not valid')
+// Koa specific product 
+router.get('/product/:id', async ctx => {
+  const product = await Product.findOne({ _id: ctx.params.id })
 
-  if (errors.length) {
-    req.flash('errors', { msg: errors.join('<br>') })
-    return res.redirect('/admin/product/' + id)
+  ctx.render('admin/product', { product })
+})
+
+//end of Koa specific product
+
+
+// view/edit product 
+// router.get('/product/:id', function (req, res) {
+//   var id = req.params.id
+//   Product.findOne({ _id: id }, function (err, product) {
+//     res.render('admin/product', {
+//       product
+//     })
+//   })
+// })
+
+
+
+
+// add new/edit
+// router.post('/product', function (req, res) {
+//   var id = req.body._id
+//   var body = req.body
+
+//   var errors = []
+//   if (!validator.isCurrency(body.price))
+//     errors.push('Price is not valid')
+
+//   if (errors.length) {
+//     req.flash('errors', { msg: errors.join('<br>') })
+//     return res.redirect('/admin/product/' + id)
+//   }
+
+//   async.waterfall([
+//     function (callback) {
+//       if (body._id.length) {
+//         Product.findOne({ _id: body._id }, function (err, product) {
+//           product = _.merge(product, req.body)
+//           callback(null, product)
+//         })
+//       } else {
+//         delete body._id //remove empty id from user
+
+//         var product = new Product(body)
+//         callback(null, product)
+//       }
+//     },
+
+//     function (product, callback) {
+//       product.save(function (err, saved) {
+//         callback(err, saved)
+//       })
+//     }
+//   ], function (err, product) {
+//     if (err) {
+//       console.log(err)
+//       req.flash('errors', [{ msg: err.message }])
+//       return res.redirect('/admin/product?' + qs.stringify(req.body))
+//     }
+
+//     req.flash('success', [{ msg: product.name + ' saved' }])
+//     res.redirect('/admin/products')
+//   })
+// })
+
+/****Koa product edit/view   *******/
+router.post('/product', async (ctx, next) => {
+  const body = ctx.request.body
+  
+  if (ctx.errors) {
+    ctx.flash('errors', ctx.errors)
+    return ctx.redirect('/admin/product?', qs.stringify(body))
   }
```

Comment:
given that we are not running validations here, please remove this section. the entire `if (ctx.errors)` block